docs/dev_guide/feathr_overall_release_guide.md

```diff
@@ -0,0 +1,19 @@
+# When to Release
+- For each major and minor version release, please follow these steps. 
+- For patch versions, there should be no releases.
+
+# Writing Release Note
```

Comment:
Should how to generate/update CHANGELOG covered in this release checklist doc?

Reply:
fixed.